src/svg_surface.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright 2015-2016, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::ffi::CString;
+use std::ops::Deref;
+use std::path::Path;
+
+extern crate libc;
+use libc::{c_char, c_double};
+
+use ffi;
+use ffi::enums::SurfaceType;
+
+use surface::{Surface, SurfaceExt};
+
+#[cfg(feature = "use_glib")]
+use glib::translate::*;
+
+pub struct SVGSurface(Surface);
+
+extern "C" {
+    pub fn cairo_svg_surface_create (filename: *const c_char,
+                                     width_in_points: c_double,
+                                     height_in_points: c_double) -> *mut ffi::cairo_surface_t;
+}
+
+impl SVGSurface {
+    pub fn from(surface: Surface) -> Result<SVGSurface, Surface> {
+        if surface.get_type() == SurfaceType::Svg {
+            Ok(SVGSurface(surface))
+        } else {
+            Err(surface)
+        }
+    }
+
+    #[doc(hidden)]
+    pub unsafe fn from_raw_full(ptr: *mut ffi::cairo_surface_t) -> SVGSurface {
+        Self::from(Surface::from_raw_full(ptr)).unwrap()
+    }
+
+    pub fn create<T: AsRef<Path>>(filename: T, width: f64, height: f64) -> SVGSurface
+    {
+        // Convert: AsRef<Path> -> Cow<str> -> str
+        let s = filename.as_ref().to_string_lossy().into_owned();
```

Comment:
Lossy does not seem ideal. Also does Cairo always expect UTF8 on all platforms? If not this should go through `OsStr`. `Path` can be converted to `OsStr`.

Reply:
i'm not really sure how to fix this